include/pru/am572x.h

```diff
@@ -0,0 +1,769 @@
+};
+
+
+struct __PRU_INTC_SICR_S {
+	volatile unsigned int STATUS_CLR_INDEX		: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_EISR_S {
+	volatile unsigned int ENABLE_SET_INDEX		: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_EICR_S {
+	volatile unsigned int ENABLE_CLR_INDEX		: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_HIEISR_S {
+	volatile unsigned int HINT_ENABLE_SET_INDEX	: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+
+struct __PRU_INTC_HIDISR_S {
+	volatile unsigned int HINT_ENABLE_CLR_INDEX	: 10;
+	const volatile unsigned int __reserved_1	: 22;
+};
+
+struct __PRU_INTC_GPIR_S {
+	const volatile unsigned int GLB_PRI_INTR	: 10;
+	const volatile unsigned int __reserved_1	: 21;
+	const volatile unsigned int GLB_NONE		: 1;
+};
+
+
+
+struct __PRU_INTC_CMR_S {
+	volatile unsigned int CH_MAP			 : 4;
+	const volatile unsigned int __reserved_1 : 4;
+} __attribute__((packed));
+
+
+struct __PRU_INTC_HMR_S {
+	volatile unsigned int HINT_MAP			 : 4;
+	const volatile unsigned int __reserved_1 : 4;
+} __attribute__((packed));
+
+
+struct __PRU_INTC_HIPIR_S {
+	const volatile unsigned int PRI_HINT		: 10;
+	const volatile unsigned int __reserved_1	: 21;
+	const volatile unsigned int NONE_HINT		: 1;
+};
+
+
+struct __PRU_INTC_HINLR_S {
+	volatile unsigned int AUTO_OVERRIDE		 : 10;
+	const volatile unsigned int __reserved_1 : 21;
+	volatile unsigned int NEST_HINT			 : 1;
+};
+
+struct __PRU_INTC_HIER_S {
+	volatile unsigned int ENABLE_HINT		 : 10;
+	const volatile unsigned int __reserved_1 : 22;
+};
+
+struct __PRU_INTC {
+	union {
+		const volatile unsigned int REVID;
+		struct __PRU_INTC_REVID_S REVID_bit;
+	};
+	union {
+		volatile unsigned int CR;
+		struct __PRU_INTC_CR_S CR_bit;
+	};
+	unsigned int __gap1[2];
+	union {
+		volatile unsigned int GER;
+		struct __PRU_INTC_GER_S GER_bit;
+	};
+	unsigned int __gap2[2];
+	union {
+		volatile unsigned int GNLR;
+		struct __PRU_INTC_GNLR_S GNLR_bit;
+	};
+	union {
+		volatile unsigned int SISR;
+		struct __PRU_INTC_SISR_S SISR_bit;
+	};
+	union {
+		volatile unsigned int SICR;
+		struct __PRU_INTC_SICR_S SICR_bit;
+	};
+	union {
+		volatile unsigned int EISR;
+		struct __PRU_INTC_EISR_S EISR_bit;
+	};
+	union {
+		volatile unsigned int EICR;
+		struct __PRU_INTC_EICR_S EICR_bit;
+	};
+	unsigned int __gap3[1];
+	union {
+		volatile unsigned int HIEISR;
+		struct __PRU_INTC_HIEISR_S HIEISR_bit;
+	};
+	union {
+		volatile unsigned int HIDISR;
+		struct __PRU_INTC_HIEISR_S HIDISR_bit;
+	};
+	unsigned int __gap4[17];
+	union {
+		volatile unsigned int GPIR;
+		struct __PRU_INTC_GPIR_S GPIR_bit;
+	};
+	unsigned int __gap5[95];
+	volatile unsigned int SRSR0;
+	volatile unsigned int SRSR1;
+	unsigned int __gap6[30];
+	volatile unsigned int SECR0;
+	volatile unsigned int SECR1;
+	unsigned int __gap7[30];
+	volatile unsigned int ESR0;
+	volatile unsigned int ERS1;
```

Comment:
Good catch - it is ERS1 not ESR1. Can you please also fix for AM3 ?

Reply:
Fixed in e7201c47ba29ddb72374093ea299e4daa37bd9be